catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/PeriodParser.java

```diff
@@ -0,0 +1,126 @@
+      int sumOfFractionalValuesInSeconds =
+          sumOfFractionalValuesInSeconds(fullDecimalValues, unitDurationsInSeconds);
+
+      Period period =
+          new Period(
+              roundedValues[0],
+              roundedValues[1],
+              roundedValues[2],
+              roundedValues[3],
+              roundedValues[4],
+              roundedValues[5],
+              roundedValues[6],
+              0);
+
+      return period.plusSeconds(sumOfFractionalValuesInSeconds);
+    } else {
+      throw new IllegalArgumentException("Period not in valid format: " + periodText);
+    }
+  }
+
+  private static long[] getUnitDurationsInSeconds() {
+    long[] unitDurationsInSeconds = new long[7];
+    unitDurationsInSeconds[0] = ChronoUnit.YEARS.getDuration().getSeconds();
+    unitDurationsInSeconds[1] = ChronoUnit.MONTHS.getDuration().getSeconds();
+    unitDurationsInSeconds[2] = ChronoUnit.WEEKS.getDuration().getSeconds();
+    unitDurationsInSeconds[3] = ChronoUnit.DAYS.getDuration().getSeconds();
+    unitDurationsInSeconds[4] = ChronoUnit.HOURS.getDuration().getSeconds();
+    unitDurationsInSeconds[5] = ChronoUnit.MINUTES.getDuration().getSeconds();
+    unitDurationsInSeconds[6] = (long) 1; // seconds per second
+    return unitDurationsInSeconds;
+  }
+
+  private static BigDecimal[] parseFullDecimalValues(String text, Matcher matcher) {
+    BigDecimal[] fullDecimalValues = new BigDecimal[7];
+    int unitNumber;
+    String unitName;
+
+    for (TimeUnit unit : TimeUnit.values()) {
+      unitNumber = unit.ordinal();
+      unitName = unit.toString().toLowerCase();
+      fullDecimalValues[unitNumber] = parseNumber(text, matcher.group(unitNumber + 1), unitName);
+    }
+    return fullDecimalValues;
+  }
+
+  private static BigDecimal parseNumber(CharSequence text, String parsed, String errorText) {
+    if (parsed == null) {
+      return BigDecimal.ZERO;
+    }
+    try {
+      return new BigDecimal(parsed);
+    } catch (NumberFormatException ex) {
+      throw new DateTimeParseException(
+          "Text cannot be parsed to a Duration: " + errorText, text, 0, ex);
+    }
+  }
+
+  private static int[] roundValuesDownToNearestInt(BigDecimal[] fullDecimalValues) {
+    int[] roundedValues = new int[7];
+
+    for (int i = 0; i < fullDecimalValues.length; i++) {
+      roundedValues[i] = fullDecimalValues[i].setScale(0, RoundingMode.DOWN).intValue();
+    }
+    return roundedValues;
+  }
+
+  private static int sumOfFractionalValuesInSeconds(
+      BigDecimal[] fullDecimalValues, long[] unitDurationsInSeconds) {
+    BigDecimal fractionalValue;
+    BigDecimal valueInSeconds;
+    BigDecimal sum = BigDecimal.ZERO;
+
+    for (int i = 0; i < fullDecimalValues.length; i++) {
+      fractionalValue = fullDecimalValues[i].remainder(BigDecimal.ONE);
+      valueInSeconds = fractionalValue.multiply(new BigDecimal(unitDurationsInSeconds[i]));
+      sum = sum.add(valueInSeconds);
+    }
```

Comment:
❓ I know this is just a port of another PR, but I was wondiner if sum.intValue() would cause an issue if for instance a user put in 80.5 years?  That would overflow the int.

Reply:
That's a really good point. I'm going to create a follow up ticket for this.